### ISSUE TITLE
Emoji clear text limitations

### ISSUE THREAD

**Issue by nicojs** [stryker-mutator/stryker-js#3839]
**Summary**

During our hacktoberfest hackathon, we've managed to implement #3820 (thanks @brdv ♥)

However, there are still some issues:

1. Not every terminal supports emojis:
   ![image](https://user-images.githubusercontent.com/1828233/199797251-5e16e085-5a51-4c0b-92bb-052caab4b423.png)
    
1. The width of emojis is terminal dependent.
 
  ![image](https://user-images.githubusercontent.com/1828233/199794591-16db5e20-1719-4d3c-b8ae-dab18768719e.png)
  
  ```
  ------------------------------|---------|-----------|------------|-------------|-----------|-----------|
  File                          | % score | ✅ killed | ⌛️ timeout | 👽 survived | 🙈 no cov | 💥 errors |
  ------------------------------|---------|-----------|------------|-------------|-----------|-----------|
  ```
  (in the default mac terminal the width is now correct)

**Stryker config**

```json
{
  "clearTextReporter": { "allowEmojis": true }
}
```

Excellent solutions would be: 
1. Detect if the terminal supports emojis
2. Detect the width of emojis in the terminal

@JoshuaKGoldberg can you help with your all-knowing emoji skills? 🙇‍♂️

**Comment by JoshuaKGoldberg** [comment]
This is beyond what I would be able to contribute with emojis now, sorry 😞. To my knowledge most programs just make educated guesses what width emojis are, based on the emoji's character range.

**Comment by nicojs** [comment]
Not really. The console-table-printer should allow you to configure the `charLength` which doesn't seem to work as expected, or I don't understand the `charLength` feature 😅

If the `charLength` feature works, one could also imagine such a feature in Stryker, making emojis work for everyone who wants to spend the time configuring it for their console.
